c_src/membrane_shm_payload/lib.c

```diff
@@ -107,31 +102,28 @@ int shm_payload_get_from_term(ErlNifEnv * env, ERL_NIF_TERM struct_term, ShmPayl
   payload->name = enif_alloc(name_binary.size + 1);
   memcpy(payload->name, (char *) name_binary.data, name_binary.size);
   payload->name[name_binary.size] = '\0';
-  payload->name_size = name_binary.size + 1;
 
   return 1;
 }
 
 /**
  * Allocates POSIX shared memory given the data (name, capacity) in ShmPayload struct.
  *
+ * If 'retry_flag' is set to true, this function tries to regenerate name an try again in
+ * case the name for shm is already taken.
+ *
  * Shared memory can be accessed by using 'shm_payload_open_and_mmap'.
  * Memory will be unmapped when ShmPayload struct is freed (by 'shm_payload_release')
  */
-ShmPayloadLibResult shm_payload_allocate(ShmPayload * payload) {
+ShmPayloadLibResult shm_payload_allocate(ShmPayload * payload, int retry_flag) {
```

Comment:
Instead of accepting this flag, this function could simply check whether the name is null and if so try to generate it